miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/signatureverifier/SignatureVerifier.kt

```diff
@@ -72,22 +78,28 @@ internal class SignatureVerifier(
         return@withContext isVerified
     }
 
-    private fun rawToEncodedECPublicKey(key: String): ECPublicKey {
-        val parameters = ecParameterSpecForCurve("secp256r1")
-        val keySizeBytes = parameters.order.bitLength() / java.lang.Byte.SIZE
-        val pubKey = Base64.decode(key, Base64.DEFAULT)
+    private fun rawToEncodedECPublicKey(key: String): ECPublicKey? {
+        try {
+            val parameters = ecParameterSpecForCurve("secp256r1")
+            val keySizeBytes = (parameters.order.bitLength()) / java.lang.Byte.SIZE
+            val pubKey = Base64.decode(key, Base64.DEFAULT)
 
-        // First Byte represents compressed/uncompressed status
-        // We're expecting it to always be uncompressed (04)
-        var offset = UNCOMPRESSED_OFFSET
-        val x = BigInteger(POSITIVE_BIG_INTEGER, pubKey.copyOfRange(offset, offset + keySizeBytes))
+            // First Byte represents compressed/uncompressed status
+            // We're expecting it to always be uncompressed (04)
 
-        offset += keySizeBytes
-        val y = BigInteger(POSITIVE_BIG_INTEGER, pubKey.copyOfRange(offset, offset + keySizeBytes))
+            var offset = UNCOMPRESSED_OFFSET
+            val x = BigInteger(POSITIVE_BIG_INTEGER, pubKey.copyOfRange(offset, offset + keySizeBytes))
 
-        val keySpec = ECPublicKeySpec(ECPoint(x, y), parameters)
-        val keyFactory = KeyFactory.getInstance("EC")
-        return keyFactory.generatePublic(keySpec) as ECPublicKey
+            offset += keySizeBytes
+            val y = BigInteger(POSITIVE_BIG_INTEGER, pubKey.copyOfRange(offset, offset + keySizeBytes))
+
+            val keySpec = ECPublicKeySpec(ECPoint(x, y), parameters)
+            val keyFactory = KeyFactory.getInstance("EC")
+            return keyFactory.generatePublic(keySpec) as ECPublicKey
+        } catch (e: java.lang.Exception) {
```

Comment:
I have put an exception handling, that's why the other changes has been reflected. Basically, there is no change in `ECPublicKey` creation.